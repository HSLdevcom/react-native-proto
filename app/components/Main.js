/**
 * Main component
 * @flow
 */

import React, {Component} from 'react';
import {
    AppState,
    DeviceEventEmitter,
    Platform,
} from 'react-native';
import Beacons from 'react-native-beacons-manager';
import {connect} from 'react-redux';
import CustomWebView from './CustomWebView';
import {getBeaconData} from '../actions/beacons';

export const REITTIOPAS_MOCK_URL = 'https://reittiopas.fi/?mock';
export const REITTIOPAS_URL = 'https://reittiopas.fi';


/*
* Current UUID for OnyxBeacon
* 20CAE8A0-A9CF-11E3-A5E2-0800200C9A66
* Stops
* DFFF7ADA-A48A-4F77-AA9A-3A7943641E6C
*/
const beaconId = (Platform.OS === 'ios') ?
'DFFF7ADA-A48A-4F77-AA9A-3A7943641E6C' :
'dfff7ada-a48a-4f77-aa9a-3a7943641e6c';
const vehicleBeaconId = (Platform.OS === 'ios') ?
'20CAE8A0-A9CF-11E3-A5E2-0800200C9A66' :
'20cae8a0-a9cf-11e3-a5e2-0800200c9a66';
const testBeaconId = (Platform.OS === 'ios') ?
'7D7AFDB9-14A3-EECC-A67D-DBD798A33C25' :
'7d7afdb9-14a3-eecc-a67d-dbd798a33c25';

const beaconRegion = {
    identifier: 'OnyxBeacon',
    uuid: beaconId,
};

const vehicleBeaconRegion = {
    identifier: 'OnyxBeacon',
    uuid: vehicleBeaconId,
};
const testBeaconRegion = (Platform.OS === 'ios') ? {
    identifier: 'Livi',
    uuid: testBeaconId,
} : testBeaconId;

class Main extends Component { // eslint-disable-line react/prefer-stateless-function

    state = {
        appState: AppState.currentState,
    }

    componentWillMount = () => {
        if (Platform.OS === 'android') {
            Beacons.detectIBeacons();
        }
        Beacons.startMonitoringForRegion(beaconRegion);
        Beacons.startMonitoringForRegion(vehicleBeaconRegion);
        Beacons.startMonitoringForRegion(testBeaconRegion);
        if (Platform.OS === 'ios') {
            Beacons.startUpdatingLocation();
        }
        this.props.getBeaconData();
        DeviceEventEmitter.addListener(
            'regionDidEnter',
            (data) => {
                console.log('MONITORING - regionDidEnter data: ', data);
                this.props.getBeaconData();
            }
        );
        DeviceEventEmitter.addListener(
            'regionDidExit',
            (data) => {
                console.log('MONITORING - regionDidExit data: ', data);
            }
        );
    }

    // TODO: add options view and define there if user wants to use this with "?mock"?
    render() {
        return (
            <CustomWebView uri={REITTIOPAS_MOCK_URL} />
        );
    }
}

Main.propTypes = {
    getBeaconData: React.PropTypes.func.isRequired,
};

function mapStateToProps(state) {
    return {

    };
}
function mapDispatchToProps(dispatch) {
    return {
        getBeaconData: () => dispatch(getBeaconData()),
    };
}

export default connect(
     mapStateToProps,
     mapDispatchToProps
 )(Main);
