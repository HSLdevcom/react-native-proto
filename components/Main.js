/**
 * Main component
 * @flow
 */

import React, {Component} from 'react';
import {StyleSheet, View, WebView} from 'react-native';

const styles = StyleSheet.create({
    container: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
        backgroundColor: 'rgb(0, 122, 201)',
    },
    text: {
        color: 'rgb(255, 255, 255)',
        height: 50,
    },
    webView: {
        marginBottom: 50,
        marginTop: 60,
        width: '100%',
    },
});

class Main extends Component { // eslint-disable-line react/prefer-stateless-function
    render() {
        return (
            <View style={styles.container}>
                <WebView
                    style={styles.webView}
                    source={{uri: 'https://www.reittiopas.fi/'}}
                    scalesPageToFit
                />
            </View>
        );
    }
}

export default Main;
