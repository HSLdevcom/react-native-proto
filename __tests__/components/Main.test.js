// https://facebook.github.io/jest/blog/2016/07/27/jest-14.html
import 'react-native';
import React from 'react';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import Main from '../../components/Main';

it('renders correctly', () => {
    const tree = renderer.create(<Main />).toJSON();
    expect(tree).toMatchSnapshot();
});
