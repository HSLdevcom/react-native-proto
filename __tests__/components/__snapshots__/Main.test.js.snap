// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgb(0, 122, 201)",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#999999"
    hidesWhenStopped={true}
    size="large"
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "padding": 8,
        },
        Object {
          "height": 80,
          "marginTop": 100,
          "position": "absolute",
          "zIndex": 2,
        },
      ]
    }
  />
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RCTWebView
      allowsInlineMediaPlayback={undefined}
      automaticallyAdjustContentInsets={undefined}
      bounces={undefined}
      contentInset={undefined}
      dataDetectorTypes={undefined}
      decelerationRate={undefined}
      injectedJavaScript={undefined}
      mediaPlaybackRequiresUserAction={undefined}
      messagingEnabled={false}
      onLoadingError={[Function]}
      onLoadingFinish={[Function]}
      onLoadingStart={[Function]}
      onMessage={[Function]}
      onShouldStartLoadWithRequest={undefined}
      scalesPageToFit={true}
      scrollEnabled={undefined}
      source={
        Object {
          "uri": "https://www.reittiopas.fi/",
        }
      }
      style={
        Array [
          Object {
            "flex": 1,
          },
          Object {
            "backgroundColor": "#ffffff",
          },
          Object {
            "marginBottom": 50,
            "marginTop": 63,
            "width": "100%",
          },
        ]
      }
    />
  </View>
</View>
`;
